command/agent.go

```diff
@@ -116,6 +120,15 @@ func (c *AgentCommand) Flags() *FlagSets {
 			"\"trace\", \"debug\", \"info\", \"warn\", and \"err\".",
 	})
 
+	f.IntVar(&IntVar{
+		Name:    "max-connection-attempts",
+		Target:  &c.flagMaxConnectionAttempts,
+		Default: 0,
+		EnvVar:  "VAULT_MAX_CONNECTION_ATTEMPTS",
```

Comment:
Why is connection attempts preferred over a timeout as a configuration?  As a user I know was "60s" means, but I don't know how long a connection attempt takes.

Reply:
That's a fair question. The original request for this was for something akin to [`VAULT_MAX_RETRIES`][max_retries]. A timeout could likely work just as well


[max_retries]: https://www.vaultproject.io/docs/commands/index.html#vault_max_retries

Reply:
Timeout works for me!